### PR TITLE
feat: add merge-queue bot

### LINKED ISSUE #4285: Merge-on-green: maintain a merge queue to reduce number of tests run
When there are many pull requests pending merge on green and when the repository has "require up-to-date branches" enabled, there's a race for which pull request is merged first. All of the other pending branches are then updated from main and they race again to complete CI and potentially merge next. So if there are N approved and pending automerge PRs, we will run the CI suite `N * (N - 1) / 2` times (sum of 1 up to N).

### PR DESCRIPTION
**Pull request by tmatsuo**
This bot manages a single queue for each repo, trying to merge PRs in the queue in order.

fixes #4285 

related #2871

### CODE REVIEW

File: packages/merge-queue/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# merge-queue
+
+This is a small bot for managing merge queue for repositories.
```
Comment (SurferJeffAtGoogle) [inline packages/merge-queue/README.md:3]: Please mention that the moment [Github's merge queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) becomes available to us, we'll delete this bot.

File: packages/merge-queue/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## How to use
+
+Add `merge-queue:add` label to PRs you want to automatically merge. The bot will
+put the pull requests in a queue and try to merge them in order. This bot is
+especially useful if you configure pull requests need to be up to date before
+merge.
+
+
+## Deploying and testing
```
Comment (SurferJeffAtGoogle) [inline packages/merge-queue/README.md:13]: Please discuss how an attacker might attempt to use this bot to merge malicious code into the code base.

File: packages/merge-queue/README.md (hunk 1 above)
Comment (tmatsuo) [reply packages/merge-queue/README.md:3]: There is a possibility that we cannot just use Github's merge queue as a replacement. But yeah if Github provides what we want, we can delete this bot.

File: packages/merge-queue/README.md (hunk 2 above)
Comment (tmatsuo) [reply packages/merge-queue/README.md:13]: The attacker needs to have permission to add a label to a PR. In this regard, the situation is the same as merge-on-green.

File: packages/merge-queue/README.md (hunk 1 above)
Comment (tmatsuo) [reply packages/merge-queue/README.md:3]: Added a sentence

File: packages/merge-queue/README.md (hunk 2 above)
Comment (tmatsuo) [reply packages/merge-queue/README.md:22]: Added some notes

**Comment by tmatsuo** [comment]
@SurferJeffAtGoogle I added some sentences to the README. PTAL

**Comment by tmatsuo** [comment]
Closing until someone wants this feature

**Pull request closed without merging**